lib/IRGen/GenEnum.cpp

```diff
@@ -2610,9 +2645,15 @@ namespace {
     void assignWithTake(IRGenFunction &IGF, Address dest, Address src,
                         SILType T)
     const override {
-      if (IGF.isInOutlinedFunction() || (TIK < Loadable) ||
-          (T.hasArchetype())) {
+      if (IGF.isInOutlinedFunction()) {
         emitIndirectAssign(IGF, dest, src, T, IsTake);
+      } else if (T.hasArchetype()) {
+        llvm::SmallVector<std::pair<CanType, llvm::Value *>, 4>
+            typeToMetadataVec;
+        collectArchetypeMetadata(IGF, typeToMetadataVec, T);
+        IGF.IGM.generateCallToGenericOutlinedCopyAddr(
+            IGF, *TI, dest, src, typeToMetadataVec, T,
+            &IRGenModule::getOrCreateGenericOutlinedAssignWithTakeFunction);
       } else {
         // Create an outlined function to avoid explosion
         IGF.IGM.generateCallToOutlinedCopyAddr(
```

Comment:
Why don't you merge the generic and non-generic cases, where in the non-generic case there are no additional metadata arguments?

Reply:
Will do that / update the PR